### ISSUE TITLE
df.loc does not work as expected

### ISSUE THREAD

**Issue by ayushdas** [modin-project/modin#2892]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS Linux 7 (Core)
- **Modin version**: 0.8.3
- **Python version**: 3.6.8
- **Code we can use to reproduce**:  # Works with pandas
import pandas as pd
d = {'col1': [1, 2], 'col2': [3, 4], 'col3': [5,6]}
df = pd.DataFrame(d)
df = df.set_index(['col1', 'col2'])
df.loc[[1,2]]

# Does not work with modin pandas
import modin.pandas as pd
d = {'col1': [1, 2], 'col2': [3, 4], 'col3': [5,6]}
df = pd.DataFrame(d)
df = df.set_index(['col1', 'col2'])
df.loc[[1,2]]

### Describe the problem
The df.loc function does not work in Modin pandas multi index Dataframe as expected, as the behaviour does not match with the pandas .loc functionality (refer code snippet for reproducing error).

### Source code / logs
UserWarning: Distributing <class 'dict'> object. This may take some time.

KeyErrorTraceback (most recent call last)
/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2897             try:
-> 2898                 return self._engine.get_loc(casted_key)
   2899             except KeyError as err:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()

KeyError: 2

The above exception was the direct cause of the following exception:

KeyErrorTraceback (most recent call last)
<ipython-input-34-1a411248228a> in <module>
     11 df = pd.DataFrame(d)
     12 df = df.set_index(['col1', 'col2'])
---> 13 df.loc[[1,2]]

/usr/local/lib/python3.6/site-packages/modin/pandas/indexing.py in __getitem__(self, key)
    526                 return self.df.iloc[:, slice(*result_slice)]
    527 
--> 528         row_lookup, col_lookup = self._compute_lookup(row_loc, col_loc)
    529         if any(i == -1 for i in row_lookup) or any(i == -1 for i in col_lookup):
    530             raise KeyError(

/usr/local/lib/python3.6/site-packages/modin/pandas/indexing.py in _compute_lookup(self, row_loc, col_loc)
    676                 row_lookup = self.qc.index.get_indexer_for(row_loc)
    677             else:
--> 678                 row_lookup = self.qc.index.get_locs(row_loc)
    679         elif is_boolean_array(row_loc):
    680             # If passed in a list of booleans, we return the index of the true values

/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/multi.py in get_locs(self, seq)
   3100                 indexer = _update_indexer(
   3101                     _convert_to_indexer(
-> 3102                         self.get_loc_level(k, level=i, drop_level=False)[0]
   3103                     ),
   3104                     indexer=indexer,

/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/multi.py in get_loc_level(self, key, level, drop_level)
   2883                 return indexer, maybe_mi_droplevels(indexer, ilevels, drop_level)
   2884         else:
-> 2885             indexer = self._get_level_indexer(key, level=level)
   2886             return indexer, maybe_mi_droplevels(indexer, [level], drop_level)
   2887 

/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/multi.py in _get_level_indexer(self, key, level, indexer)
   2967         else:
   2968 
-> 2969             code = self._get_loc_single_level_index(level_index, key)
   2970 
   2971             if level > 0 or self.lexsort_depth == 0:

/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/multi.py in _get_loc_single_level_index(self, level_index, key)
   2635             return -1
   2636         else:
-> 2637             return level_index.get_loc(key)
   2638 
   2639     def get_loc(self, key, method=None):

/usr/local/lib64/python3.6/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2898                 return self._engine.get_loc(casted_key)
   2899             except KeyError as err:
-> 2900                 raise KeyError(key) from err
   2901 
   2902         if tolerance is not None:

KeyError: 2

**Comment by devin-petersohn** [comment]
Hi @ayushdas, thanks for reporting!

It looks like you are using Modin 0.8.3, but the latest release is 0.9.1. Would you be able to update and check if the issue is resolved for you? Thanks again for the report!

**Comment by ayushdas** [comment]
Hi @devin-petersohn I have tried pip install modin==0.9.1 but got the following error 

ERROR: Could not find a version that satisfies the requirement modin==0.9.1 (from versions: 0.0.2, 0.1.0, 0.1.1, 0.1.2, 0.2.0, 0.2.2, 0.2.3, 0.2.4, 0.2.5, 0.3.0rc1, 0.3.0, 0.3.1, 0.4.0rc1, 0.4.0, 0.5.0rc1, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.7.4, 0.8.0, 0.8.1, 0.8.1.1, 0.8.2, 0.8.3)
ERROR: No matching distribution found for modin==0.9.1

**Comment by ayushdas** [comment]
@devin-petersohn I have just checked with modin 0.9.1. The above error still persists in this version as well.

**Comment by YarShev** [comment]
Yes, the issue is reproducible on master.

**Comment by gshimansky** [comment]
Yes, in Modin parser of `loc` expressions for MultiIndex frames breaks on some of them because it is not as robust as in Pandas. As in bug https://github.com/modin-project/modin/issues/2576 there is a workaround for this bug. You can try to use expression like this:
```python
df.loc[slice(1, 2)]
```
and it works.

**Comment by pyrito** [comment]
I am still able to reproduce this error on the latest master. Seems like an issue related to MultiIndex.
